src/Internal.AspNetCore.CodeSign.Abstractions/SignRequestManifestXmlReader.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.BuildTools.CodeSign
+{
+    /// <summary>
+    /// Reads a sign request manifest xml fil.
+    /// </summary>
+    public class SignRequestManifestXmlReader
+    {
+        private readonly TextReader _reader;
+        private readonly string _manifestBasePath;
+
+        /// <summary>
+        /// Load the sign request xml file.
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <returns></returns>
+        public static SignRequestCollection Load(string filePath)
+        {
+            using (var reader = File.OpenText(filePath))
+            {
+                return Load(reader, Path.GetDirectoryName(filePath));
+            }
+        }
+
+        /// <summary>
+        /// Read the sign request xml using a given base path.
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="manifestBasePath"></param>
+        /// <returns></returns>
+        public static SignRequestCollection Load(TextReader reader, string manifestBasePath)
+        {
+            return new SignRequestManifestXmlReader(reader, manifestBasePath).Read();
+        }
+
+        private SignRequestManifestXmlReader(TextReader reader, string manifestBasePath)
+        {
+            _reader = reader;
+            _manifestBasePath = manifestBasePath;
+        }
+
+        private SignRequestCollection Read()
+        {
+            var doc = XDocument.Load(_reader);
+
+            var collection = new SignRequestCollection
+            {
+                BasePath = _manifestBasePath
+            };
+
+            foreach (var element in doc.Root.Elements())
+            {
+                collection.Add(ReadElement(element));
+            }
+
+            return collection;
+        }
+
+        private SignRequestItem ReadElement(XElement element)
+        {
+            switch (element.Name.ToString())
+            {
+                case "Nupkg":
+                    return ReadNupkg(element);
+                case "Vsix":
+                    return ReadVsix(element);
+                case "Zip":
+                    return ReadZip(element);
+                case "File":
+                    return SignRequestItem.CreateFile(GetPath(element), GetCertificate(element), GetStrongName(element));
```

Comment:
Nitpick: personally I would create a `ReadFile` function just to follow the pattern we did with Nupkg Vsix etc. Total preference though, I see that it's less complicated than the others.